### ISSUE TITLE
recommended way to add mruby gems

### ISSUE THREAD

**Issue by yannick** [h2o/h2o#993]
what is the recommended way to add more mruby gems?

and is there a way to update mruby itself? 
could this be made a git submodule so it can be switched?

**Comment by kazuho** [comment]
For the purpose, I would suggest doing the following. `mruby` directory under the build directory will be recreated automatically.
* remove `mruby` directory under your build directory (I assume you are doing partially right here)
* dump an mrbgem under `deps`, preserving the `mruby-` prefix (see https://github.com/h2o/h2o/blob/master/misc/mruby_config.rb#L16-L25 to find out how the mrbgems are noticed)
* run `make` with appropriate libraries added to `-DEXTRA_LIBS`

Especially the last step makes me uncomfortable, but fixing this is a low-priority issue since we do not recommend install mrbgems anyways.

The reason we do not recommend is because some of them _might_ block the event loop of H2O for a certain amount of time (e.g. sqlite3 blocked by `fsync`, redis client blocked by network access).

**Comment by yannick** [comment]
ok thank you @kazuho , i was unsure if there was a more appropriate way.
i am aware of the blocking issue, but for certain use cases it might still be ok.
in the long run i guess it would be nice to have a more pluggable way and an ecosystem of selected gems that work with the eventloop. 
one use case i see is to build something similar to https://github.com/Mashape/kong 

unless you want to tag it FAQ you can close the issue.
